### PR TITLE
Added user label support in sql_database_instance (#1372)

### LINKED ISSUE #1372: Support label in sql_database_instance
The API has a field called `settings.userLabels` to store a dict of key-value user-provided labels.

See: https://cloud.google.com/sql/docs/mysql/admin-api/v1beta4/instances

cc/ @stefanthorpe

### PR DESCRIPTION
**Pull request by ortaman**
Fixes #1372

### CODE REVIEW

**Comment by ortaman** [comment]
To enhancement: #1372

File: google/resource_sql_database_instance.go [hunk 1]
```diff
@@ -595,6 +601,15 @@ func resourceSqlDatabaseInstanceCreate(d *schema.ResourceData, meta interface{})
 		settings.ReplicationType = v.(string)
 	}
 
+	settings.UserLabels = map[string]string{}
+
+	if v, ok := _settings["user_labels"]; ok {
```
Comment (rosbo) [inline google/resource_sql_database_instance.go]: Use [convertStringMap](https://github.com/terraform-providers/terraform-provider-google/blob/master/google/utils.go#L267)

File: google/resource_sql_database_instance.go [hunk 2]
```diff
@@ -1027,6 +1042,14 @@ func resourceSqlDatabaseInstanceUpdate(d *schema.ResourceData, meta interface{})
 			settings.ReplicationType = v.(string)
 		}
 
+		if v, ok := _settings["user_labels"]; ok {
```
Comment (rosbo) [inline google/resource_sql_database_instance.go]: Same, user convertStringMap

File: google/resource_sql_database_instance_test.go [hunk 3]
```diff
@@ -521,6 +521,33 @@ func TestAccSqlDatabaseInstance_multipleOperations(t *testing.T) {
...
+	var instance sqladmin.DatabaseInstance
+	databaseID := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccSqlDatabaseInstanceDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: fmt.Sprintf(
```
Comment (rosbo) [inline google/resource_sql_database_instance_test.go]: Can you add a new step in this test to test the update path for labels?

File: google/resource_sql_database_instance.go (hunk 1 above)
Comment (ortaman) [reply google/resource_sql_database_instance.go]: done

File: google/resource_sql_database_instance_test.go (hunk 3 above)
Comment (ortaman) [reply google/resource_sql_database_instance_test.go]: done

**Pull request merged**